production.py

```diff
@@ -263,13 +263,18 @@ def do(cls, productions):
             InternalShipment.assign_try(shipments)
 
     def get_cost(self, name):
+        pool = Pool()
+        Uom = pool.get('product.uom')
+
         price = super().get_cost(name)
-        purchase = self.purchase_request and self.purchase_request.purchase
-        if not purchase:
+        line = self.purchase_request and self.purchase_request.purchase_line
+        if not line:
             return price
-
-        return price + purchase.untaxed_amount
-
+        
+        quantity = Uom.compute_qty(
+            self.uom, self.quantity, self.product.default_uom)
+        
+        return price + Decimal(quantity) * line.unit_price
```

Comment:
In case purchase line, the unit is "kg" and the production move is "gr" (different units, same category), the unit price from purchase line, required to convert to "default uom" product before to add price in the cost.

Reply:
Right, but instead of converting the unit_price to "default uom" wouldn't it be simpler to just add the line.product.cost_price?